Add gate-featured Canon usage/derivation.

CPerezz opened the pull request:
Since we don't necessarily need `Canon` derived for
our structures. Therefore, a gate-featured solution for
canonnical and it's deps is the best solution.

Therefore, `Canon` is not derived for any of the crate exported
data structures unless we use the `canon` feature.

Commit:
Update jubjub & poseidon deps

Commit:
Derive `Canon` for `StealthAddress`

Commit:
Update dep versions of poseidon & Jubjub

ZER0 reviewed (changes requested):
Update the CI in order to be consistent with our other repos. As discussed you might want to put `Canon` behind a feature flag.

Commit:
Add gate-featured canonical usage/derivation

Since we don't necessarily need `Canon` derived for
our structures. Therefore, a gate-featured solution for
canonnical and it's deps is the best solution.

Therefore, `Canon` is not derived for any of the crate exported
data structures unless we use the `canon` feature.

- Add canon feature to test
- Bump Poseidon252, plonk & jubjub deps

CPerezz commented:
> Update the CI in order to be consistent with our other repos. As discussed you might want to put `Canon` behind a feature flag.

Both done. Tell me what you think.

Commit:
Bump to v0.3.0

Commit:
Rustfmt

ZER0 commented on Cargo.toml:
```diff
@@ -16,3 +18,6 @@ thiserror = "1.0"
 default-features = false
+
+[features]
+canon = ["canonical", "canonical_derive", "dusk-jubjub/canon", "poseidon252/canon"]
```
Nit: add a new line

Commit:
Address nits
